### ISSUE TITLE
input_dims not a constant expression in slice_op.cc 

### ISSUE THREAD

**Issue by ejls** [tensorflow/tensorflow#14324]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Debian 9.0 (with a Gentoo Prefix)
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: c81acfb025abe417d80ffa677edfe2dd1bcda58e (15 hours old)
- **Python version**: 3.6.3
- **Bazel version (if compiling from source)**: 0.6.1
- **GCC/Compiler version (if compiling from source)**: 6.4.0
- **CUDA/cuDNN version**: no CUDA
- **GPU model and memory**: no GPU
- **Exact command to reproduce**: bazel build --config=mkl --config=opt //tensorflow/tools/pip_package:build_pip_package

### Describe the problem
It seems that 8011eda4b7 merged a C++ error in file [tensorflow/core/kernels/slice_op.cc line 255](https://github.com/tensorflow/tensorflow/blob/db430c4bb275e05ee65dc27c7f771f778bfcbcf2/tensorflow/core/kernels/slice_op.cc#L255).
As seen [line 232](https://github.com/tensorflow/tensorflow/blob/db430c4bb275e05ee65dc27c7f771f778bfcbcf2/tensorflow/core/kernels/slice_op.cc#L232), input_dims is a constant (const), but is not a constant expression (constexpr), thus it should not be used as a template parameter.

### Source code / logs
```
ERROR: /local/esimon/prefix/usr/local/src/tensorflow/tensorflow/core/kernels/BUILD:780:1: C++ compilation of rule '//tensorflow/core/kernel$
:slice_op' failed (Exit 1).
tensorflow/core/kernels/slice_op.cc: In member function 'void tensorflow::MklSliceOp<Device, T>::Compute(tensorflow::OpKernelContext*)':
tensorflow/core/kernels/slice_op.cc:255:35: error: the value of 'input_dims' is not usable in a constant expression
         functor::Slice<Device, T, input_dims>()(
                                   ^~~~~~~~~~
tensorflow/core/kernels/slice_op.cc:232:15: note: 'input_dims' was not initialized with a constant expression
     const int input_dims = input.dims();
               ^~~~~~~~~~
tensorflow/core/kernels/slice_op.cc:255:45: error: the value of 'input_dims' is not usable in a constant expression
         functor::Slice<Device, T, input_dims>()(
                                             ^
tensorflow/core/kernels/slice_op.cc:232:15: note: 'input_dims' was not initialized with a constant expression
     const int input_dims = input.dims();
               ^~~~~~~~~~
tensorflow/core/kernels/slice_op.cc:255:45: note: in template argument for type 'int'
         functor::Slice<Device, T, input_dims>()(
                                             ^
```

**Comment by bignamehyp** [comment]
@gunan @drpngx @aselle , can you please take a look?
It might be related to https://github.com/tensorflow/tensorflow/pull/11140

**Comment by ejls** [comment]
It seems to have been fixed in d0a5d885
I'm closing.
